satellite/accounting/rollup/rollup.go

```diff
@@ -64,56 +67,62 @@ func (r *Service) Close() error {
 // Rollup aggregates storage and bandwidth amounts for the time interval.
 func (r *Service) Rollup(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	// only Rollup new things - get LastRollup
-	lastRollup, err := r.sdb.LastTimestamp(ctx, accounting.LastRollup)
-	if err != nil {
-		return Error.Wrap(err)
-	}
-	// unexpired orders with created at times before the last rollup timestamp could still have been added later
-	if !lastRollup.IsZero() {
-		lastRollup = lastRollup.Add(-r.OrderExpiration)
-	}
 
-	rollupStats := make(accounting.RollupStats)
-	latestTally, err := r.RollupStorage(ctx, lastRollup, rollupStats)
+	nodeIDs, err := r.overlay.GetAllNodeIDs(ctx)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
-	err = r.RollupBW(ctx, lastRollup, rollupStats)
-	if err != nil {
-		return Error.Wrap(err)
-	}
+	for _, id := range nodeIDs {
+		nodeLastRollup, err := r.sdb.NodeLastTimestamp(ctx, id, accounting.LastRollup)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+		// unexpired orders with created at times before the last rollup timestamp could still have been added later
+		if !nodeLastRollup.IsZero() {
+			nodeLastRollup = nodeLastRollup.Add(-r.OrderExpiration)
+		}
 
-	// remove the latest day (which we cannot know is complete), then push to DB
-	latestTally = time.Date(latestTally.Year(), latestTally.Month(), latestTally.Day(), 0, 0, 0, 0, latestTally.Location())
-	delete(rollupStats, latestTally)
-	if len(rollupStats) == 0 {
-		r.logger.Info("RollupStats is empty")
-		return nil
-	}
+		rollupStats := make(accounting.RollupStats)
+		latestTally, err := r.RollupStorage(ctx, id, nodeLastRollup, rollupStats)
+		if err != nil {
+			return Error.Wrap(err)
+		}
 
-	err = r.sdb.SaveRollup(ctx, latestTally, rollupStats)
-	if err != nil {
-		return Error.Wrap(err)
-	}
+		err = r.RollupBW(ctx, id, nodeLastRollup, rollupStats)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+
+		// remove the latest day (which we cannot know is complete), then push to DB
+		latestTally = time.Date(latestTally.Year(), latestTally.Month(), latestTally.Day(), 0, 0, 0, 0, latestTally.Location())
+		delete(rollupStats, latestTally)
+		if len(rollupStats) == 0 {
+			r.logger.Info("RollupStats is empty")
+			return nil
+		}
 
-	if r.deleteTallies {
-		// Delete already rolled up tallies
-		latestTally = latestTally.Add(-r.OrderExpiration)
-		err = r.sdb.DeleteTalliesBefore(ctx, latestTally)
+		err = r.sdb.SaveRollup(ctx, latestTally, rollupStats)
 		if err != nil {
 			return Error.Wrap(err)
 		}
-	}
 
+		if r.deleteTallies {
+			// Delete already rolled up tallies
+			latestTally = latestTally.Add(-r.OrderExpiration)
+			err = r.sdb.DeleteTalliesBefore(ctx, latestTally)
```

Comment:
DeleteTalliesBefore should probably be storagenode specific - this could cause issues